### PR TITLE
[batch] fix issues with exit code, duration

### PR DESCRIPTION
**Pull request by cseed**
This fixes a few of problems:
 - Batch.jobs() was discarding the job json objects and returning an Job objects.  This is not OK because we need the metadata from the json objects for efficiency (e.g. in CI PR/batch pages).
 - We confused Job.status(), which is the job json object, with the status field of that object.  In particular, we got that wrong in BatchFormatVersion.get_status_exit_code_duration in the call to Job.{exit_code, total_duration_msecs}.  Add a test to check we got this right.
 - Return the duration in msecs, not a string, because as an API, the string is useless.  humanize it before sending it to the API.

I think the batch client is (1) has a lot of legacy baggage that can be cleaned up, and (2) is massive overkill and can probably be cleaned up.  It should probably have no objects, pass around job_id ints or (batch_id, job_id) tuples, and be a thin wrapper around request calls.  Pipeline should be the high level API.

Finally, I think we need a little document (just a comment) about the format of GET /batches/batch_id and GET /jobs/job_id.  It's changed enough that it is getting a little hard to follow.

I'm going to dev deploy to test the UI.  Don't approve until it checks out, please.

FYI @jigold

### CODE REVIEW

**Comment by danking** [comment]
looks fine to me

**Comment by cseed** [comment]
OK, everything checks out in dev deploy.
